src/services/refactors/inlineVariable.ts

```diff
@@ -0,0 +1,236 @@
+import {
+    cast,
+    Debug,
+    Diagnostics,
+    emptyArray,
+    Expression,
+    factory,
+    FindAllReferences,
+    getExpressionPrecedence,
+    getLocaleSpecificMessage,
+    getSynthesizedDeepClone,
+    getTokenAtPosition,
+    Identifier,
+    InitializedVariableDeclaration,
+    isCallLikeExpression,
+    isExportAssignment,
+    isExportModifier,
+    isExportSpecifier,
+    isExpression,
+    isFunctionLike,
+    isIdentifier,
+    isInitializedVariable,
+    isTypeQueryNode,
+    isVariableDeclarationInVariableStatement,
+    isVariableStatement,
+    needsParentheses,
+    Node,
+    Program,
+    refactor,
+    some,
+    SourceFile,
+    SymbolFlags,
+    textChanges,
+    textRangeContainsPositionInclusive,
+    TypeChecker,
+    VariableDeclaration,
+} from "../_namespaces/ts";
+import { RefactorErrorInfo, registerRefactor } from "../_namespaces/ts.refactor";
+
+const refactorName = "Inline variable";
+const refactorDescription = getLocaleSpecificMessage(Diagnostics.Inline_variable);
+
+const inlineVariableAction = {
+    name: refactorName,
+    description: refactorDescription,
+    kind: "refactor.inline.variable"
+};
+
+interface InliningInfo {
+    references: Node[];
+    declaration: VariableDeclaration;
+    replacement: Expression;
+}
+
+registerRefactor(refactorName, {
+    kinds: [inlineVariableAction.kind],
+
+    getAvailableActions(context) {
+        const {
+            file,
+            program,
+            preferences,
+            startPosition,
+            triggerReason
+        } = context;
+
+        // tryWithReferenceToken is true below when triggerReason === "invoked", since we want to
+        // always provide the refactor in the declaration site but only show it in references when
+        // the refactor is explicitly invoked.
+        const info = getInliningInfo(file, startPosition, triggerReason === "invoked", program);
+        if (!info) {
+            return emptyArray;
+        }
+
+        if (refactor.isRefactorErrorInfo(info) && preferences.provideRefactorNotApplicableReason) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [{
+                    ...inlineVariableAction,
+                    notApplicableReason: info.error
+                }]
+            }];
+        }
+
+        if (!refactor.isRefactorErrorInfo(info)) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [inlineVariableAction]
+            }];
+        }
+
+        return emptyArray;

```

Comment:
Move the success case earlier.

```suggestion
        if (!refactor.isRefactorErrorInfo(info)) {
            return [{
                name: refactorName,
                description: refactorDescription,
                actions: [inlineVariableAction]
            }];
        }

        if (preferences.provideRefactorNotApplicableReason) {
            return [{
                name: refactorName,
                description: refactorDescription,
                actions: [{
                    ...inlineVariableAction,
                    notApplicableReason: info.error
                }]
            }];
        }
        return emptyArray;
```